File: 485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.py
# Last updated: 3/25/2026, 9:05:13 AM
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        max = count = 0
        n = len(nums)

        for i in range(n):
            if(nums[i] == 1): 
                count += 1
            else:
                count = 0

            if count > max:
                max = count

        return max

File: 485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.cpp
// Last updated: 3/25/2026, 9:05:18 AM
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int cnt = 0;
        int maxCnt = INT_MIN;

        int n = nums.size();
        for(int i = 0; i < n; i++) {
            if(nums[i] == 1){
                cnt++;
            }
            else {
                cnt = 0;
            }
            maxCnt = max(maxCnt, cnt);
        }
        return maxCnt;
    }
};